cmd/flags_core.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+
+	"github.com/celestiaorg/celestia-node/node"
+)
+
+var (
+	coreRemoteF = "core.remote"
+)
+
+func CoreFlags() *flag.FlagSet {
+	flags := &flag.FlagSet{}
+
+	flags.String(
+		coreRemoteF,
+		"",
+		"Indicates node to connect to the given remote core node. "+
+			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
+	)
+
+	return flags
+}
+
+func ParseCoreFlags(cmd *cobra.Command, env *Env) error {
+	coreRemote := cmd.Flag(coreRemoteF).Value.String()
+	if coreRemote != "" {
+		proto, addr, err := parseAddress(coreRemote)
+		if err != nil {
+			return fmt.Errorf("cmd: while parsing '%s': %w", coreRemoteF, err)
+		}
+
+		env.addOption(node.WithRemoteCore(proto, addr))
+	}
+
+	return nil
+}
+
+// parseAddress parses the given address of the remote core node
+// and checks if it configures correctly
+func parseAddress(address string) (string, string, error) {
+	u, err := url.Parse(address)
+	if err != nil || u.Scheme == "" || u.Host == "" {
+		return "", "", err
+	}
```

Comment:
if `u.Scheme` is empty but err == nil, it'll still return a nil error even though you want it to return an err right?

Reply:
That's the code @vgonkivs wrote to parse an address and its is correct.

Reply:
That's a code that was originally written by @vgonkivs and I think it is correct enough.

Reply:
If the correct address is passed but without Scheme or Host we just return. Though we can make it even more restrictive and error out if Host or Scheme are missing, as both are needed.

Reply:
Then I'd suggest to rename it to `validateAddress` since now it's not just parsing but also includes additional conditions for validation.

Reply:
Errored